### PR TITLE
Add helper script to update distro packaging hashes

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Let's not merge something that doesn't work for all distributions. I'm also not fond of the pygit2 dependency. Let's just use git for this. Shelling out really isn't that bad for stuff like this, you automatically get proper error reporting as git will handle that for you instead of having to start catching exceptions and whatnot from pygit2. It's also one less dependency that people need to install on their systems.

**Comment by keszybz** [comment]
OK. I updated this to shell out.

**Comment by keszybz** [comment]
I don't know how to make the comment better. For rpm-based, we could do `rpmautospec generate-changelog` and take the first entry. And for deb-based, read the changelog file?

File: tools/update-distro-hash.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        '--no-fetch',
+        dest='fetch',
+        action='store_false',
+        default=True,
+    )
+    return p.parse_args()
+
+def read_config(distro: str):
+    _, images = mkosi.config.parse_config(['-d', distro])
+    return images[-1]
```
Comment (daandemeyer) [inline tools/update-distro-hash.py]: Can you please use `mkosi --json summary` instead so we don't depend on internal API?

**Comment by keszybz** [comment]
Updated to use `mkosi --json summary`. I also added some logging to make it easier to see what is going on.

I didn't change the commit messages. If somebody thinks it's important, we can always add that later. We currently get a hash and the top commit title, which should be enough to identify the commit.

**Comment by keszybz** [comment]
Actually, I think including a `log --pretty=oneline` is a good idea. PTAL at the last two commits.

**Review by DaanDeMeyer — APPROVED** [review]

**Review by bluca — APPROVED** [review]

**Pull request merged**